pycalphad/model.py

```diff
@@ -855,6 +949,80 @@ def atomic_ordering_energy(self, dbe):
         return ordered_energy - ordered_energy.xreplace(molefraction_dict)
 
 
+    # TODO: fix case for VA interactions: L(PHASE,A,VA:VA;0)-type parameters
+    def shift_reference_state(self, reference_states, dbe, contrib_mods=None, output=('GM', 'HM', 'SM', 'CPM'), fmt_str="{}R"):
+        """
+        Add new attributes for calculating properties w.r.t. an arbitrary pure element reference state.
+
+        Parameters
+        ----------
+        reference_states : Iterable of ReferenceState
+            Pure element ReferenceState objects. Must include all the pure
+            elements defined in the current model.
+        dbe : Database
+            Database containing the relevant parameters.
+        output : Iterable, optional
+            Parameters to subtract the ReferenceState from, defaults to ('GM', 'HM', 'SM', 'CPM').
+        contrib_mods : Mapping, optional
```

Comment:
Is this functionality (`contrib_mods`) tested anywhere?

Reply:
It's currently tested in ESPEI 😉 I can write one for pycalphad's test suite.